### DISCUSSION TITLE
Render with Parameters GUI (similar to .Rmd Knit with Parameters)

### DISCUSSION THREAD

**Discussion by DOH-TJB0303** [quarto-dev/quarto-cli#7581] [Feature Requests]
### Description

My team and a lot of other public health agencies have built and utilize pre-built `.Rmd` reporting templates that allow end-users to quickly and easily explore data using the `Knit with Parameters GUI` available in `.Rmd` files. It would be great this this GUI functionality could be integrated within Quarto (as it was with R Markdown). 

Although parameters can be specified through `quarto::render()` and other knitting functions, the GUI is a relied upon tool to support no/low-code users which may not be confident/comfortable to edit pre-built code templates (but are able to select options they want to look at).

If implemented, the GUI would remove a major barrier to transitioning many of our reporting templates from `.Rmd` to `.Qmd`, thank you!

**Comment by mcanouil** [comment]
This repository is about Quarto CLI not editors.

An example using Quarto CLI using parameters (see <https://quarto.org/docs/computations/parameters.html>):

````qmd
---
title: "Lipsum"
params:
  amazing-content: "This is amazing content"
engine: knitr
---

`{r} params[["amazing-content"]]`
````

```sh
quarto render index.qmd --execute-param "amazing-content:actually not really"
```

or providing parameters as a YAML file

```yaml
amazing-content: "Yes it is amazing"
```

```sh
quarto render index.qmd --execute-params params.yml
```

For feature request regarding:
- RStudio: <https://github.com/rstudio/rstudio>
- VSCode Quarto extension: <https://github.com/quarto-dev/quarto>.

**Comment by cderv** [comment]
Just to be thorough regarding the R Markdown ecosystem, the "the Knit with Parameters GUI available in .Rmd files" is not an IDE thing. This is possible at command line using `rmarkdown::render("MyDocument.Rmd", params = "ask")`
Docs at: https://bookdown.org/yihui/rmarkdown/params-knit.html#the-interactive-user-interface

The interface is built with Shiny [**miniui**](https://github.com/rstudio/miniUI) - `rmarkdown::render()` does know how to run a shiny app, and run it before calling `rmarkdown::render()`. This is possible because it is all R from start to finish

With quarto it is a bit different, and less integrated, but it is possible to do the same:
 * Build a UI that would run to get user input
 * Call a rendering with quarto using parameters 
 
I don't know if we'll add that in `quarto` - maybe **quarto** R package could have something like that but there is no plan for it. 

Any webapp would work to get input, serialize to YAML and pass it to Quarto render. Taking into account specificity of parameters for Quarto  (no R object can be passed except being serialize - e.g. you can't pass a data.Frame as parameter). 

Anyhow, current solution is to develop its own interface IMO to wrap call to Quarto.

**Reply by DOH-TJB0303** [reply]
Thanks @cderv for that helpful explanation, and exploring the feasibility of such a feature -- I know it would be really valuable for a lot of folks that I work and collaborate with. 

I had recently transitioned this issue over to the RStudio IDE GitHub (https://github.com/rstudio/rstudio/issues/13937), but it sounds like I should transition it again over to the Quarto R package repository:  https://github.com/quarto-dev/quarto-r.

**Comment by rosericazondekon** [comment]
Dear @DOH-TJB0303 ,

You might want to look up the [Knit with Parameters](https://open-vsx.org/extension/rosericazondekon/knit-with-parameters) extension in your Positron/VS Code extension view.

It brings the familiar “Knit with Parameters” GUI workflow to Positron and supports both R Markdown and Quarto. It also goes a bit further by supporting parameter discovery and evaluation for both R and Python.

I hope you find it useful!

**Reply by mcanouil** [reply]
@rosericazondekon thanks, but please avoid resurrecting 3 years old closed topics, especially posting the same thing multiple times.
There is a "Show & Tell" section to advertise tools.
